MapboxDirections/MBVisualInstruction.swift

```diff
@@ -2,127 +2,99 @@ import Foundation
     
     /**
-     :nodoc:
-     Initialize a `VisualInstruction` from a dictionary given a `DrivingSide`.
+     The degrees at which you will be exiting a roundabout, assuming 180 indicates going straight through the roundabout.
      */
-    @objc public convenience init(json: [String: Any], drivingSide: DrivingSide) {
-        let distanceAlongStep = json["distanceAlongGeometry"] as! CLLocationDistance
-        
-        let primaryTextComponent = json["primary"] as! JSONDictionary
-        let primaryText = primaryTextComponent["text"] as! String
-        let primaryManeuverType = ManeuverType(description: primaryTextComponent["type"] as! String) ?? .none
-        let primaryManeuverDirection = ManeuverDirection(description: primaryTextComponent["modifier"] as! String)  ?? .none
-        let primaryTextComponents = (primaryTextComponent["components"] as! [JSONDictionary]).map {
-            VisualInstructionComponent(maneuverType: primaryManeuverType, maneuverDirection: primaryManeuverDirection, json: $0)
-        }
-        
-        var secondaryText: String?
-        var secondaryTextComponents: [VisualInstructionComponent]?
-        if let secondaryTextComponent = json["secondary"] as? JSONDictionary {
-            secondaryText = secondaryTextComponent["text"] as? String
-            let secondaryManeuverType = ManeuverType(description: secondaryTextComponent["type"] as! String) ?? .none
-            let secondaryManeuverDirection = ManeuverDirection(description: secondaryTextComponent["modifier"] as! String)  ?? .none
-            secondaryTextComponents = (secondaryTextComponent["components"] as! [JSONDictionary]).map {
-                VisualInstructionComponent(maneuverType: secondaryManeuverType, maneuverDirection: secondaryManeuverDirection, json: $0)
-            }
-        }
-        
-        self.init(distanceAlongStep: distanceAlongStep, primaryText: primaryText, primaryTextComponents: primaryTextComponents, secondaryText: secondaryText, secondaryTextComponents: secondaryTextComponents, drivingSide: drivingSide)
-    }
+    @objc public var degrees: CLLocationDegrees = 180
     
     /**
      :nodoc:
-     Initialize a `VisualInstruction`.
+     Initialize A `VisualInstructionBanner`.
      */
-    @objc public init(distanceAlongStep: CLLocationDistance, primaryText: String, primaryTextComponents: [VisualInstructionComponent], secondaryText: String?, secondaryTextComponents: [VisualInstructionComponent]?, drivingSide: DrivingSide) {
-        self.distanceAlongStep = distanceAlongStep
-        self.primaryText = primaryText
-        self.primaryTextComponents = primaryTextComponents
-        self.secondaryText = secondaryText
-        self.secondaryTextComponents = secondaryTextComponents
-        self.drivingSide = drivingSide
+    @objc public init(text: String?, maneuverType: ManeuverType, maneuverDirection: ManeuverDirection, textComponents: [VisualInstructionComponent], degrees: CLLocationDegrees = 180) {
```

Comment:
Is a default degree necessary?

Reply:
If it wasn't for ObjC bridging, I'd prefer `degrees: CLLocationDegrees? = nil` but 180 seems appropriate for default value as it indicates `.straightAhead`.